### PR TITLE
[stable/traefik] Add HTTP-01 challenge support

### PR DESCRIPTION
**Pull request by todoesverso**
## What does this PR do?
This PR allows users to generate/renew ACME certificates using the HTTP-01 challenge.

## Motivation
Due to a security vulnerability, Let's Encrypt disabled TLS-SNI-01 challenges.
More information [here](https://community.letsencrypt.org/t/2018-01-09-issue-with-tls-sni-01-and-shared-hosting-infrastructure/49996) and [here](https://community.letsencrypt.org/t/2018-01-11-update-regarding-acme-tls-sni-and-shared-hosting-infrastructure/50188).

## Additional Notes
 * In order for this to work the image 1.5.0-rc5 or higger is needed.

### CODE REVIEW

File: stable/traefik/README.md [hunk 1]
```diff
@@ -107,6 +107,7 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `acme.email`                    | Email address to be used in certificates obtained from Let's Encrypt | `admin@example.com`                       |
 | `acme.staging`                  | Whether to get certs from Let's Encrypt's staging environment        | `true`                                    |
 | `acme.logging`                  | display debug log messages from the acme client library              | `false`                                   |
+| `acme.httpChallenge`                  | enables HTTP-01 challenge to generate the certificates              | `false`                                   |
```
Comment (krancour) [inline stable/traefik/README.md]: There is a PR open to add a DNS challenge. See #1299.

Taking this into consideration and assuming that _other_ types of challenges could be supported in the future as well, we shouldn't use a boolean for opting into one particular sort of challenge. Instead, we should use a discriminator that identifies _which_ of many possible challenge types should be used.

In [this comment](https://github.com/kubernetes/charts/pull/1299#issuecomment-360379019) I have proposed using `acme.challengeType`. I propose the same here.

File: stable/traefik/values.yaml [hunk 2]
```diff
@@ -34,6 +34,13 @@ acme:
   email: admin@example.com
   staging: true
   logging: false
+  ## Enable httpChallenge http://v1-5.archive.docs.traefik.io/configuration/acme/#acmehttpchallenge
+  ## Only available in Traefik >=1.5.
+  ## To use it change:
+  ##  imageTag: 1.5
```
Comment (krancour) [inline stable/traefik/values.yaml]: Wouldn't it be better to just bump the Traefik version that this chart uses by default?

File: stable/traefik/README.md (hunk 1 above)
Comment (todoesverso) [reply stable/traefik/README.md]: I can do this change no problem.

File: stable/traefik/values.yaml (hunk 2 above)
Comment (todoesverso) [reply stable/traefik/values.yaml]: Yes, the thing is that when I created this PR 1.5 was still a RC, but it was relaeased 2 day ago. Let me do those changes,

**Comment by johnhamelink** [comment]
@todoesverso please update to Traefik 1.5.1, which includes a bugfix on the new ACME support in Traefik!

**Comment by todoesverso** [comment]
@johnhamelink done :+1:

**Comment by turbopape** [comment]
Hey guys, can you also please update the readme about this issue ? I for one have been struggling with example.com certificates and not understanding what's going on :) 
Thank you for the great chart by the way :) 👍

**Comment by krancour** [comment]
#1299 was recently merged and overlaps with this one quite a bit. (It added support for the DNS challenge, but that change, necessarily included support for selecting the challenge type.) This PR is great, but needs to be rebased on top of that recent work. I think it will actually reduce this PR to just a few lines.

**Comment by todoesverso** [comment]
Ok, let me do the changes. Thanks!

**Comment by kachkaev** [comment]
@todoesverso there is a new similar PR #3838, which also adds data collection option. That PR does not need rebasing and it's been more active recently, so it has a slightly higher chance to be merged (at least at the time of writing). If you are about to rebase this PR, please hold on a bit 😉

**Comment by krancour** [comment]
Since this PR is still in need of rebase, going to close this in favor of #3838, which is the same change + some and merge-ready. Thank you for the effort on this @todoesverso.

/close

**Comment by todoesverso** [comment]
Sorry I could not do the rebase, hadn't had a free slot in days. Glad the feature is merged thou!

**Pull request closed without merging**